community/cypher/cypher-compiler-3.0/src/main/scala/org/neo4j/cypher/internal/compiler/v3_0/commands/expressions/DistanceFunction.scala

```diff
@@ -87,12 +105,12 @@ object HaversinCalculator extends DistanceCalculator {
 
   private val EARTH_RADIUS_METERS = 6378140.0
 
-  override def isDefinedAt(p1: Geometry, p2: Geometry): Boolean =
+  override def isDefinedAt(p1: Point, p2: Point): Boolean =
     p1.crs == CRS.WGS84 && p2.crs == CRS.WGS84
 
-  override def calculateDistance(p1: Geometry, p2: Geometry): Double = {
-    val c1: Seq[Double] = p1.coordinates.map(toRadians)
-    val c2: Seq[Double] = p2.coordinates.map(toRadians)
+  override def calculateDistance(p1: Point, p2: Point): Double = {
+    val c1: Seq[Double] = p1.coordinate.values.map(toRadians)
```

Comment:
`c1` and `c2` are accessed by index and should therefore either be `Array[Double]` or `Vector[Double]`